src/main/java/cohen/maze/Maze.java

```diff
@@ -0,0 +1,166 @@
+package cohen.maze;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Random;
+
+public class Maze {
+    private Cell[][] maze;
+    private int height;
+    private int width;
+
+    //generate the full 2d array of cell objects. all the 'visited's are false and all the walls are true(that's done in the cells constructor).
+    public Maze(int height, int width) {
+        //initialize the number of rows and columns in the maze with the numbers entered
+        this.height = height;
+        this.width = width;
+
+        //initialize the 2d maze with the number of spaces allocated by the height and width entered
+        maze = new Cell[height][width];
+        //fill the maze with cells
+        generateMaze();
+        //dig a an open path through the maze
+        generatePath();
+        //print the maze in the console
+        printMaze();
+    }
+
+    /**
+     * The generateMaze method fills the maze with unvisited, walled cells.
+     */
+    public void generateMaze() {
+        for(int i = 0; i < height; i++) {
+            for(int j = 0; j < width; j++) {
+                maze[i][j] = new Cell(i, j);
+            }
+        }
+    }
+
+    /**
+     * The generatePath method digs or generates a random open path through the maze of cells.
+     */
+    private void generatePath() {
+        //Find the starting point to start digging.
+        Random x = new Random();
```

Comment:
create one Random object, make it a constant. It can be used for everything

Reply:
But then the x and y coordinates would have to be the same?

Reply:
jokessss... forgot how it worked for a second, never mind!